### PR TITLE
WEB-126: 'cypress' configuration and testing/automation harness

### PR DESCRIPTION
**Pull request by atla5**
Jira Ticket: [WEB-126](https://bulibrary.atlassian.net/browse/WEB-126)

### Background:
- it's valuable to set up some additional automation to ensure our addons are working as intended, for internal and external use
- in order to do this, we need to be able to repeatably/reliably/programmatically run the dev environment from scratch 
- this will eventually (hopefully) be extended and pay off in running our other custom views going forward.

### Description of Changes
- add `npm scripts` to `package.json` for scheduling the steps in the pipeline
- create `./scripts/*.sh` for executing the more complicated pieces
- `npm install` cypress, register, and configure, using those pieces
- create a basic smoke test to serve as an example for future tests

### CODE REVIEW

**Comment by atla5** [comment]
running into some odd and under-explained errors running this

former error showing up when running locally and trying to access the page
```
Error: socket hang up
    at createHangUpError (_http_client.js:330:15)
    at Socket.socketOnEnd (_http_client.js:431:23)
    at Socket.emit (events.js:202:15)
    at endReadableNT (_stream_readable.js:1129:12)
    at processTicksAndRejections (internal/process/next_tick.js:76:17)
```

this more recent one is happening directly after running it at the command line:
```
/root/project/_build/node_modules/require-dir/index.js:97
            if (!require.extensions.hasOwnProperty(ext)) {
                                    ^

TypeError: require.extensions.hasOwnProperty is not a function
    at requireDir (/root/project/_build/node_modules/require-dir/index.js:97:37)
    at Object.<anonymous> (/root/project/_build/gulpfile.js:5:1)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at execute (/usr/local/lib/node_modules/gulp/node_modules/gulp-cli/lib/versioned/^3.7.0/index.js:28:18)
```

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            - ~/.npm
+            - ~/.cache
+      # set up the devenv/'_build' directory
+      - run:
+          command: npm run reset
+      # create the view and start server before starting tests
+      - run:
+          command: npm run restart
+          background: true
+      - run: sleep 25 && npm run cy:run
```
Comment (etgrieco) [inline circle.yml]: I use a utility script that checks that the server is alive to handle this!

`wait-for.sh`
```sh
wget --waitretry=5 --retry-connrefused -vO- $1 > /dev/null
```

then

```sh
./wait_for.sh http://localhost:8003/primo-explore/search && npm run cy:run
```

Comment (atla5) [reply circle.yml]: i tried using [`wait-on`](https://www.npmjs.com/package/wait-on) very [briefly](https://github.com/bulib/primo-explore-bu/pull/11/commits/19ad801d57ab9398d13fb6af5cb688ec42642f13), but have pushed that off to remove the extra complexity until i get green.

thank you, though, this is definitely better than the manual sleep. 👍

**Comment by atla5** [comment]
switching to 'UMNLibraries/primo-explore-hathitrust-availability' method, and potentially 'GitHub Actions'.

**Pull request closed without merging**